### PR TITLE
scaffolds a data package

### PR DESCRIPTION
**Pull request by skuhlmann**
## GitHub Issue

[gh issue.](https://github.com/HausDAO/daohaus-monorepo/issues/39)

## Changes

- nx gen of the data package
- setup the folder strucuture
- a few basic utils and types
- 2 classes (Haus and Query)
- Init tests

## Packages Added

Brief list of any packages added, and if folks need to rerun `yarn install` to run locally

## Checks

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

**Comment by skuhlmann** [comment]
this is the issue, i opened with the wrong one:
https://github.com/HausDAO/daohaus-monorepo/issues/39

**Review by alexkeating — COMMENTED** [review]
This is something that can probably be done in a separate pr, but it would be probably good to generate the graphql types at some point.

File: libs/data/src/utils/queries.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    }
+  }
+`;
+
+export const DAO_PROPOSALS = `
+  query proposals($dao: String!) {
+    proposals(
+      where: {dao: $dao}
+      orderBy: createdAt
+      orderDirection: desc
```
[the remark this answers is not in the document]
Comment (skuhlmann) [reply libs/data/src/utils/queries.ts:133]: Yeah good feedback - let's refine these in the next PR associated to this task:

https://github.com/HausDAO/daohaus-monorepo/issues/40

File: libs/data/src/utils/constants.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { KeychainList } from '..';
+
+export const ENDPOINTS: KeychainList = {
+  V3_SUBGRAPH: {
+    '0x4': 'https://api.thegraph.com/subgraphs/name/hausdao/daohaus-v3-rinkeby',
+    '0x64': '',
+  },
+};
```
[the remark this answers is not in the document]
Comment (skuhlmann) [reply libs/data/src/utils/constants.ts:8]: does `as const` - making these read only suffice?

```
export const ENDPOINTS: KeychainList = {
  V3_SUBGRAPH: {
    '0x4': 'https://api.thegraph.com/subgraphs/name/hausdao/daohaus-v3-rinkeby',
    '0x64': '',
  },
} as const;
```

or are you thinking these become a private class variable on the Query class. I could see that.

I added too much stuff in here for just a scaffold task. Let's figure out the best approach for this as part of the next ticket:
https://github.com/HausDAO/daohaus-monorepo/issues/40

[the remark this answers is not in the document]
Comment (alexkeating) [reply libs/data/src/utils/constants.ts]: Okay sounds good

**Review by alexkeating — APPROVED** [review]

**Pull request merged**